packages/auth/src/components/EmailPasswordSignUpForm/index.js

```diff
@@ -0,0 +1,120 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Input } from 'react-rainbow-components';
+import { FormattedMessage, useIntl } from 'react-intl';
+import { showAppMessage, showAppSpinner, hideAppSpinner } from '../../../../app/src/actions';
+import useFirebaseApp from '../../../../firebase-hooks/src/useFirebaseApp';
+
+const EmailPasswordSignUpForm = (props) => {
+    const intl = useIntl();
+    const { className, style } = props;
+    const [name, setName] = useState('');
+    const [email, setEmail] = useState('');
+    const [password, setPassword] = useState('');
+    const app = useFirebaseApp();
+    const { nameIcon, emailIcon, passwordIcon } = props;
+
+    const onSubmit = async (event) => {
+        event.preventDefault();
+
+        try {
+            showAppSpinner();
+            await app.auth().createUserWithEmailAndPassword(email, password);
+            hideAppSpinner();
+        } catch (error) {
+            hideAppSpinner();
+            showAppMessage({
+                message: `${error}`,
+                variant: 'error',
+            });
+        }
+    };
+
+    const nameLabel = <FormattedMessage id="EmailPasswordSignUpForm.name" defaultMessage="Name" />;
+    const emailLabel = (
+        <FormattedMessage id="EmailPasswordSignUpForm.email" defaultMessage="Email Address" />
+    );
+    const passwordLabel = (
+        <FormattedMessage id="EmailPasswordSignUpForm.password" defaultMessage="Password" />
+    );
+    const buttonLabel = (
+        <FormattedMessage id="EmailPasswordSignUpForm.signup" defaultMessage="Create account" />
+    );
+    const namePlaceholder = intl.formatMessage({
+        id: 'EmailPasswordSignUpForm.namePlaceholder',
+        defaultMessage: 'Enter your name',
+    });
+    const emailPlaceholder = intl.formatMessage({
+        id: 'EmailPasswordSignUpForm.emailPlaceholder',
+        defaultMessage: 'Enter your email',
+    });
+    const passwordPlaceholder = intl.formatMessage({
+        id: 'EmailPasswordSignUpForm.passwordPlaceholder',
+        defaultMessage: 'Enter your password',
+    });
+
+    const buttonStyle = {
+        width: '100%',
+    };
+
+    return (
+        <form className={className} style={style} onSubmit={onSubmit}>
+            <Input
+                label={nameLabel}
+                icon={nameIcon}
+                placeholder={namePlaceholder}
+                name="name"
+                value={name}
+                onChange={(event) => setName(event.target.value)}
+                required
+            />
+            <br />
```

Comment:
it is better to use styles instead of creating dom element to separate components